src/unit_tests/shared/test_agent_op.c

```diff
@@ -152,6 +153,65 @@ static void test_create_sendsync_payload(void **state) {
     cJSON_Delete(payload);
 }
 
+static void test_get_agent_group(void **state) {
+    char* group;
+    int agent_id = 1;
+    int wdb_sock = -1;
+    size_t size = OS_SIZE_65536 + 1;
+    int res;
+
+    os_calloc(size, sizeof(char), group);
+    snprintf(group, size, "%s","default");
+
+    cJSON *agent_info_correct = cJSON_CreateArray();
+    cJSON *agent_info = cJSON_CreateObject();
+    cJSON_AddStringToObject(agent_info, "group", group);
+    cJSON_AddItemToArray(agent_info_correct, agent_info);
+
+    cJSON *agent_info_no_field = cJSON_CreateArray();
+    cJSON *agent_info_ver = cJSON_CreateObject();
+    cJSON_AddStringToObject(agent_info_ver, "version", "1.2");
+    cJSON_AddItemToArray(agent_info_no_field, agent_info_ver);
+
+    cJSON *agent_info_several_correct = cJSON_CreateArray();
+    cJSON *agent_info_several_ok = cJSON_CreateObject();
+    cJSON_AddStringToObject(agent_info_several_ok, "group", "default,Group1");
+    cJSON_AddItemToArray(agent_info_several_correct, agent_info_several_ok);
+
+    /* Unable to get agent info */
+    expect_value(__wrap_wdb_get_agent_info, id, agent_id);
+    will_return(__wrap_wdb_get_agent_info, NULL);
+    expect_string(__wrap__mdebug1, formatted_msg, "Unable to get agent info of agent '1'");
+    res = get_agent_group(agent_id, group, size, NULL);
+
+    assert_int_equal(res, OS_INVALID);
+
+    /* Unable to get agent group  */
+    expect_value(__wrap_wdb_get_agent_info, id, agent_id);
+    will_return(__wrap_wdb_get_agent_info, agent_info_no_field);
+    res = get_agent_group(agent_id, group, size, NULL);
+
+    assert_int_equal(res, OS_INVALID);
+
+    /* Correct single group  */
+    expect_value(__wrap_wdb_get_agent_info, id, agent_id);
+    will_return(__wrap_wdb_get_agent_info, agent_info_correct);
+    res = get_agent_group(agent_id, group, size, NULL);
+    assert_string_equal("default", group);
+    assert_int_equal(res, OS_SUCCESS);
+
+    snprintf(group, size, "%s","default,Group2");
+
+    /* return OK, Wrong groups string  */
```

Comment:
I´m not sure if this test is required. We are verifying if that output is being over-written?

Reply:
Indeed, I was just checking that the correct json response was being written but It doesn't add anything to the other tests. I'll remove it.